models/categoryModel.test.js

```diff
@@ -0,0 +1,44 @@
+import mongoose from 'mongoose';
+import { MongoMemoryServer } from 'mongodb-memory-server';
+import Category from './categoryModel';
+
+let mongoServer;
+
+describe('Category Model', () => {
+    beforeAll(async () => {
+        mongoServer = await MongoMemoryServer.create();
+        const uri = mongoServer.getUri();
+        await mongoose.connect(uri, { useNewUrlParser: true, useUnifiedTopology: true });
+    });
+
+    afterAll(async () => {
+        await mongoose.disconnect();
+        await mongoServer.stop();
+    });
+
+    test('should create and save a category succesfully', async () => {
+        const categoryData = { name: 'Test', slug: 'test-slug' };
+        const validCategory = new Category(categoryData);
+        const savedCategory = await validCategory.save();
+
+        expect(savedCategory._id).toBeDefined();
+        expect(savedCategory.name).toBe(categoryData.name);
+        expect(savedCategory.slug).toBe(categoryData.slug);
+    });
+
+    // NEVER PASS
+    test('should throw error if name is missing', async () => {
```

Comment:
Why should it throw an error if `name` is missing? I thought in `categoryModel.js`, both `name` and `slug` are not compulsory/required attributes?

Reply:
For this I feel that it would be an error when name does not exists as you would just be adding an empty value into the database which does not make sense to me.

Reply:
However the schema did not state that these fields are `required`, so it would not throw an error?

![image](https://github.com/user-attachments/assets/f65cd758-2d3f-4ac7-9dc6-08d52fabafa1)

Reply:
Thoughts? @Aishwarya-Hariharan-Iyer @Kang-Quan @kohkaixun